samcli/local/docker/container.py

```diff
@@ -94,6 +96,24 @@ def create(self):
             "tty": False
         }
 
+        if self._delve_path:
+            LOG.info("in golang debug mode")
+            kwargs["volumes"] = {
+                self._host_dir: {
+                    # Mount the host directory as "read only" directory inside container at working_dir
+                    # https://docs.docker.com/storage/bind-mounts
+                    # Mount the host directory as "read only" inside container
+                    "bind": self._working_dir,
+                    "mode": "ro"
+                },
+                self._delve_path: {
+                    "bind": "/var/debug",
+                    "mode": "ro"
+                }
+            }
+            kwargs["security_opt"] = ["seccomp:unconfined"]
+            kwargs["cap_add"] = ["SYS_PTRACE"]
```

Comment:
This is the bit that @bmoffatt helped me understand. Can you add comments explaining what these are and why they are necessary?